File: socialview-autocomplete/src/main/java/com/hendraanggrian/appcompat/widget/HashtagArrayAdapter.java
package com.hendraanggrian.appcompat.widget;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.PluralsRes;

import com.hendraanggrian.appcompat.socialview.Hashtagable;
import com.hendraanggrian.appcompat.socialview.autocomplete.R;

/**
 * Default adapter for displaying hashtag in {@link SocialAutoCompleteTextView}. Note that this
 * adapter is completely optional, any adapter extending {@link android.widget.ArrayAdapter} can be
 * attached to {@link SocialAutoCompleteTextView}.
 */
public class HashtagArrayAdapter<T extends Hashtagable> extends SocialArrayAdapter<T> {
  private final int countPluralRes;

  public HashtagArrayAdapter(@NonNull Context context) {
    this(context, R.plurals.posts);
  }

  public HashtagArrayAdapter(@NonNull Context context, @PluralsRes int countPluralRes) {
    super(context, R.layout.socialview_layout_hashtag, R.id.socialview_hashtag);
    this.countPluralRes = countPluralRes;
  }

  @NonNull
  @Override
  @SuppressWarnings("unchecked")
  public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
    final ViewHolder holder;
    if (convertView == null) {
      convertView =
          LayoutInflater.from(getContext())
              .inflate(R.layout.socialview_layout_hashtag, parent, false);
      holder = new ViewHolder(convertView);
      convertView.setTag(holder);
    } else {
      holder = (ViewHolder) convertView.getTag();
    }
    final T item = getItem(position);
    if (item != null) {
      holder.hashtagView.setText(item.getId());

      if (item.getCount() > 0) {
        holder.countView.setVisibility(View.VISIBLE);
        final int count = item.getCount();
        holder.countView.setText(
            getContext().getResources().getQuantityString(countPluralRes, count, count));
      } else {
        holder.countView.setVisibility(View.GONE);
      }
    }
    return convertView;
  }

  private static class ViewHolder {
    private final TextView hashtagView;
    private final TextView countView;

    ViewHolder(View itemView) {
      hashtagView = itemView.findViewById(R.id.socialview_hashtag);
      countView = itemView.findViewById(R.id.socialview_hashtag_count);
    }
  }
}

File: socialview-commons/src/com/hendraanggrian/appcompat/widget/HashtagArrayAdapter.kt
package com.hendraanggrian.appcompat.widget

import android.content.Context
import android.view.LayoutInflater.from
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.annotation.PluralsRes
import com.hendraanggrian.appcompat.socialview.Hashtagable
import com.hendraanggrian.appcompat.socialview.commons.R

/**
 * Default adapter for displaying hashtag in [SocialAutoCompleteTextView].
 * Note that this adapter is completely optional, any adapter extending
 * [android.widget.ArrayAdapter] can be attached to [SocialAutoCompleteTextView].
 */
class HashtagArrayAdapter<T : Hashtagable> @JvmOverloads constructor(
    context: Context,
    @PluralsRes private val countPlural: Int = R.plurals.posts
) : SocialArrayAdapter<T>(
    context,
    R.layout.socialview_layout_hashtag,
    R.id.socialview_hashtag
) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val holder: ViewHolder
        var view = convertView
        when (view) {
            null -> {
                view = from(context).inflate(R.layout.socialview_layout_hashtag, parent, false)
                holder = ViewHolder(view!!)
                view.tag = holder
            }
            else -> holder = view.tag as ViewHolder
        }
        getItem(position)?.let { hashtag ->
            holder.hashtagView.text = hashtag.hashtag
            if (hashtag.count != null) {
                holder.countView.visibility = View.VISIBLE
                holder.countView.text =
                    context.resources.getQuantityString(countPlural, hashtag.count!!, hashtag.count)
            } else {
                holder.countView.visibility = View.GONE
            }
        }
        return view
    }

    private class ViewHolder(itemView: View) {
        val hashtagView: TextView = itemView.findViewById(R.id.socialview_hashtag)
        val countView: TextView = itemView.findViewById(R.id.socialview_hashtag_count)
    }
}
